Add integration tests for plan selection

guillaumelamirand opened the pull request:
## Description

Add integration tests for plan selection :

- Only keyless
- Only  api-key
- Only jwt
- Only oauth2
- keyless + api-key
- keyless + api-key + jwt
- keyless + api-key + jwt + oauth2 

## Additional context

<!-- Add any other context about the PR here -->
<!-- It can be links to other PRs or docs or drawing -->
<!-- Or reproduction steps in case of bug fix -->

<!-- Storybook placeholder -->
---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-rpoeuyyulr.chromatic.com)
<!-- Storybook placeholder end -->

Commit:
tests: implement single plan integration tests

Commit:
tests: add multi plan selection integration tests

leleueri reviewed (commented):
LGTM but what should be the behaviour if multiple JWT plan are configured ?

guillaumelamirand commented:
> LGTM but what should be the behaviour if multiple JWT plan are configured ?

This will be part of another PR...which is coming; adding +400tests was enough for 1 PR I suppose 😄

Commit:
fix: properly handle token extraction

   - no Authorization header will not return any security token
   - no Bearer security schema value will not return a security token (Authorization: Basic 123456)
   - Empty Bearer or apikey will return an invalid security token
   - Bearer token without client_id will now return an invalid Security Token

Commit:
fix: clean any security token if selection rule doesn't match
